src/configure/configurers/azurePipelineConfigurer.ts

```diff
@@ -103,6 +103,10 @@ export class AzurePipelineConfigurer implements Configurer {
             telemetryHelper.setTelemetry(TelemetryKeys.NewOrganization, inputs.isNewOrganization.toString());
         }
         catch (error) {
+            if ((error.typeKey as string).toUpperCase() === constants.ExceptionType.UnauthorizedRequestException) {
+                extensionVariables.isErrorWhitelisted = true;
```

Comment:
this is tricky Vineet. you are whitelisting an error by setting a global variable. What if this error is suppressed by the caller (down the stack) and another error is thrown which is not expected. Now what will happen is that the 2nd error will get whitelisted. 
I think the intention here is to say that this error is resulting in a failure which should be whitelisted. There are two ways which comes to my mind currently:
1. Rename to `isFailureWhitelisted` and ensure that execution ends with this error. Essentially we are saying that this error is resulting into a failure. I am not sure how we will enforce this.
2. Extend error object to include a variable `isExpected`. This changes the scope from global to local. Now whenever some throws an error they can decide if this is an expected error. At top level, we will check this property of error object and whitelist accodingly.

I do not have a clear preference. But I think 2nd one is bit better as it has lesser chance of whitelisting being missed/misused.

Reply:
Yeah, I also thought earlier about 2nd approach but didn't found any scenario in our code where the error is suppressed by the caller. Yeah, I will implement this approach.

Reply:
done